docs/design/satellite-svc-separation.md

```diff
@@ -0,0 +1,208 @@
+# Satellite Service Separation
+
+## Table of Contents
+1. [Abstract](#abstract)
+2. [Background](#background): Current Satellite Design
+3. [Design](#design)
+4. [Rationale](#rationale)
+4. [Implementation](#implementation)
+5. [Open Issues](#open-issues)
+
+## Abstract
+
+The goal of the design doc is to describe the work necessary to make the Satellite horizontally scalable.
+
+## Background
+
+Currently all Satellite services run in a single binary in a single container. While this is great for development, this is bad for scaling.
+
+### Current Satellite design
+
+Currently the Satellite is a single binary made of up the following services:
+
+#### overlay
+The overlay cache maintains critical information about nodes in the network (via the nodes table in Satellite.DB). One important use of overlay is to help select which storage nodes to upload files to.
+
+#### metainfo
+Metainfo is responsible for all things related to the metainfo stored for each file on the network. The metainfo system is currently composed of the following parts: 
+
+1) metainfoDB, a key/value store with segment path names as the key and a pointer as the value, 
+
+2) metainfo loop which iterates over all the key/values in metainfoDB, 
+
+3) metainfo endpoint which creates the public RPCs for creating/deleting metainfo, 
+
+4) metainfo service.
+
+#### orders
+Orders is responsible for creating/managing orders that the satellite issues for a file upload/download. Orders are used to keep track of how much bandwidth was used to upload/download a file. This data is used to pay storage nodes for bandwidth usage and to charge the uplinks. See this [doc on the lifecycle of data](https://github.com/storj/docs/blob/master/code/payments/Accounting.md#lifecycle-of-the-data) related to accounting which includes orders.
+
+#### audit
+Audit performs audits of the storage nodes to make sure the data they store is still retrievable. The audit system is currently made up of an audit service that runs on an interval performing audits on a segment at a time. The result of the audits are reported to the overlay service to store in node table in Satellite.DB. See [docs on audit](https://github.com/storj/docs/blob/master/code/audits/audit-service.md) for more details.
+
+#### repair
+Repair loops over the metainfoDB and checks if a segment is injured, if it is injured, its added to the repair queue and a repairer later fixes it. If its unable to be repaired then its added to irreparabledb. The irreparable loop later iterates over this table and attempts to re-repair these injured segments. The repair system is currently made of 4 parts and 2 DBs (db tables). The 4 parts are 1) repair observer (contains ReliabilityCache) 2) irreparable loop 3) repairer 4) repair queue. The 2 DBs are 1) injuredsegment (repair queue) table in satellite.DB 2) and irreparabledb table in satellite.DB 
```

Comment:
Repair searches metainfo for injured segments and adds them to the repair queue. Repairer picks segments from queue and tries to fix them. When repair fails, then the segment is added to irreparable database.